src/main/java/org/candlepin/resource/EnvironmentResource.java

```diff
@@ -399,7 +399,9 @@ public AsyncJobStatusDTO promoteContent(@Verify(Environment.class) String envId,
         Set<String> contentIds;
         try {
             contentIds = this.batchCreate(contentToPromote, environment);
-            this.contentAccessManager.syncOwnerLastContentUpdate(environment.getOwner());
+            // TODO: Remove the line below once certificate generation based on the `lastContentUpdate` field
```

Comment:
Whoops... these slipped through the cracks. Neither should exist anymore: batchCreate is doing an env sync already here, and in demoteContent we have an env sync after the try/catch.

Reply:
Good catch! It looks like we missed removing these in the feature branch:

https://github.com/candlepin/candlepin/blob/feature/sca_component_generation_optimizations/src/main/java/org/candlepin/resource/EnvironmentResource.java#L445-L447

Reply:
So, this exposed an issue with the `ContentAccessSpecTest.shouldShowEnvironmentContentChangeInContentAccessCert` spec tests for MySQL because MySQL does not store millisecond precision. Sometimes, the environment's last content update datetime would be in the same second as the content access payloads timestamp, but with more milliseconds in the future. So, when we check if there have been content changes, the comparison between the environment's last content update time and the content access payloads timestamp ends up being equal because of the loss of the milliseconds on the env last content update datetime.

To fix this, I had to add two sleeps for a second to guarantee that the environment's last conet

Reply:
Wouldn't that have been a problem even with that extra org.syncLastContentUpdate in place?

Reply:
Yeah, that's a good point. Maybe the `environment.getOwner().syncLastContentUpdate()` call was far enough in the call stack that it usually was being set in the next second in comparison to the content access payloads timestamp, but that is just my best guess. I wouldn't be surprised if this test was occasionally failing/flaky.